api/app/GraphQL/Validators/DigitalContractingQuestionnaireInput.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\GraphQL\Validators;
+
+use Database\Helpers\DirectiveFormsApiEnums;
+use Illuminate\Validation\Rule;
+use Nuwave\Lighthouse\Validation\Validator;
+
+final class DigitalContractingQuestionnaireInput extends Validator
+{
+    /**
+     * Return the validation rules.
+     *
+     * @return array<string, array<mixed>>
+     */
+    public function rules(): array
+    {
+        return [
+            'readPreamble' => ['accepted'],
+            'departmentOther' => [
+                'required_without:department',
+                Rule::prohibitedIf(!empty($this->arg('department')))
+            ],
+            'businessOwnerEmail' => ['email'],
+            'financialAuthorityEmail' => ['email'],
+            'authorityInvolvedOther' => [
+                Rule::requiredIf(in_array(DirectiveFormsApiEnums::CONTRACT_AUTHORITY_OTHER, $this->arg('authoritiesInvolved'))),
+                Rule::prohibitedIf(!in_array(DirectiveFormsApiEnums::CONTRACT_AUTHORITY_OTHER, $this->arg('authoritiesInvolved')))
+            ],
```

Comment:
It seems as though this arg has the potential to be `null` so when checking if it is in the array, an error is thrown rather than returning a `boolean`.

I think this is true of other items where we are checking for the existence of an item in an array. Though, I'm not sure if these are intentionally nullable or not.

Reply:
Great catch!  I worked up some custom rules (thanks for the help BTW) to make sure the primary and detail fields are consistent and handle nulls OK.